Source/Projections/Store/Projections.cs

```diff
@@ -0,0 +1,119 @@
+        readonly IPerformMethodCalls _caller;
+        readonly IResolveCallContext _callContextResolver;
+        readonly ExecutionContext _executionContext;
+        readonly IProjectionAssociations _projectionAssociations;
+        readonly IConvertProjectionsToSDK _toSDK;
+        readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Projections"/> class.
+        /// </summary>
+        /// <param name="caller">The <see cref="IPerformMethodCalls" />.</param>
+        /// <param name="callContextResolver">The <see cref="IResolveCallContext" />.</param>
+        /// <param name="executionContext">The <see cref="ExecutionContext" />.</param>
+        /// <param name="projectionAssociations">The <see cref="IProjectionAssociations" />.</param>
+        /// <param name="toSDK">The <see cref="IConvertProjectionsToSDK" />.</param>
+        /// <param name="logger">The <see cref="ILogger" />.</param>
+        public Projections(
+            IPerformMethodCalls caller,
+            IResolveCallContext callContextResolver,
+            ExecutionContext executionContext,
+            IProjectionAssociations projectionAssociations,
+            IConvertProjectionsToSDK toSDK,
+            ILogger logger)
+        {
+            _caller = caller;
+            _callContextResolver = callContextResolver;
+            _executionContext = executionContext;
+            _projectionAssociations = projectionAssociations;
+            _toSDK = toSDK;
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task<CurrentState<TProjection>> Get<TProjection>(Key key, System.Threading.CancellationToken cancellationToken = default)
+            where TProjection : class, new()
+        {
+            var projection = _projectionAssociations.GetFor<TProjection>();
+
+            _logger.LogDebug(
+                "Getting current projection state with key {Key} for projection of {ProjectionType} with id {ProjectionId} in scope {Scope}",
+                key,
+                typeof(TProjection),
+                projection.Identifier,
+                projection.ScopeId);
+
+            var request = new GetOneRequest
+            {
+                CallContext = _callContextResolver.ResolveFrom(_executionContext),
+                Key = key,
+                ProjectionId = projection.Identifier.ToProtobuf(),
+                ScopeId = projection.ScopeId.ToProtobuf()
+            };
+
+            var response = await _caller.Call(_getOneMethod, request, cancellationToken).ConfigureAwait(false);
+            response.Failure.ThrowIfFailureIsSet();
+
+            if (!_toSDK.TryConvert<TProjection>(response.State, out var state, out var error))
+            {
+                _logger.LogError(error, "The Runtime acknowledges that the projection state was returned, but the returned {State} could not be converted.", response.State);
```

Comment:
Maybe not the right way to do it - but something like:
```suggestion
                _logger.LogError(error, "The Runtime returned the projection state '{State}'. But it could not be converted to {typeof(TProjection)}.", response.State);
```